src/Compilers/Core/Portable/SourceGeneration/Nodes/NodeStateTable.cs

```diff
@@ -90,6 +90,26 @@ private NodeStateTable(ImmutableArray<TableEntry> states, ImmutableArray<Increme
         public int GetTotalEntryItemCount()
             => _states.Sum(static e => e.Count);
 
+        public int GetTotalEntryItemCountPlusEmpty()
```

Comment:
The name doesn't quite seem to match up the logic here. The logic seems to be excluding trailing empty while the name implies it's including all empty. Suggest a name like `GetTotalEntryItemCountExceptTrailingEmpty` (might be a bit wordy though).